### ISSUE TITLE
Make IndexWriter#flushNextBuffer flush deletes too?

### ISSUE THREAD

**Issue by jpountz** [apache/lucene#12572]
### Description

`IndexWriter#flushNextBuffer()` is a convenient way to control indexing buffer sizes across multiple index writers. Unfortunately, it seems that it only ever flushes DWPTs, and never deletes, even though deletes may use a very significant amount of the indexing buffer.

Can we enhance `flushNextBuffer()` to flush deletes too when appropriate?

**Comment by s1monw** [comment]
After digging into this and opening a PR for it I think this is unnecessary. I tried to beef up tests for this and this caused me to refresh my knowledge how stuff works down in the IW / DWPT. Every time we flush a DWPT we do freeze the global deletes buffer and push it to the queue. Which essentially means we are applying deletes no matter how much memory it consumes. Digging deeper I think we can / should do some cleanups in the IW regarding deletes. I did start with good / better testing and will come up with some ideas in different PRs/Issues
